python/packages/ai/teams/ai/planner/planner_error.py

```diff
@@ -0,0 +1,12 @@
+"""
+Copyright (c) Microsoft Corporation. All rights reserved.
+Licensed under the MIT License.
+"""
+
+from teams.ai.ai_error import AIError
+
+
+class PlannerError(AIError):
```

Comment:
one guide line we want to follow is do not define new errors if developer cannot handle this error programatically. Can you share how do you expect users catch and handle this error?

Reply:
Since our errors inherit they can handle by:

```python
@app.error
async def on_error(context: TurnContext, err: ApplicationError):
    if isinstance(err, PlannerError):
        .. handle planner error
        return
    .. handle normal error
```

Reply:
It also comes from Azure SDK guideline (which is summarize to the SDK guideline word I shared with you): do not create a new exception type unless the developer can handle the error programmatically.

This is a minor comment. If you think the guideline makes sense, let's adopt it. If not, we can go ahead with current implementation and discuss the guidelines further.

Reply:
Ok that works for me, In this case would we just have one error type `ApplicationError` or should we use a builtin error like `ValueError`? I am going to merge this today but during our sync we can discuss this more and create a work item to fix how we are handling errors in the codebase.

Reply:
When implementing this I was following the C# implementation but if the python way of doing things is to use builtin errors more then I'm fully behind doing it that way 👍